### PR TITLE
Unit testing for wdb_agent code

### PR DESCRIPTION
**Pull request by DProvinciani**
|Related issue|
|---|
| [5919](https://github.com/wazuh/wazuh/issues/5919) |

## Description

This pull request includes all the necessary test cases to have 100% of unit testing coverage for all the functions in the file `wdb_agent.c`.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation

- Memory tests for Linux
  - [x] Valgrind (memcheck and descriptor leaks check)

### CODE REVIEW

**Review by palaciosjeremias — APPROVED** [review]
LGTM!

File: src/unit_tests/wazuh_db/test_wdb_agent.c [hunk 1]
```diff
@@ -2954,11 +3017,13 @@ void test_wdb_get_agents_by_keepalive_wdbc_query_error(void **state) {
     int keepalive = 10;
 
     const char *query_str = "global get-agents-by-keepalive condition > 10 start_id 0";
```
Comment (pereyra-m) [inline src/unit_tests/wazuh_db/test_wdb_agent.c:3019]: Now, with the PR

File: src/unit_tests/wazuh_db/test_wdb_agent.c [hunk 2]
```diff
@@ -3287,73 +3377,72 @@ void test_wdb_update_agent_group_success(void **state)
...
     expect_value(__wrap_wdbc_query_ex, len, WDBOUTPUT_SIZE);
+    will_return(__wrap_wdbc_query_ex, response);
     will_return(__wrap_wdbc_query_ex, OS_SUCCESS);
 
     // Parsing Wazuh DB result
+    expect_any(__wrap_wdbc_parse_result, result);
     will_return(__wrap_wdbc_parse_result, WDBC_OK);
 
     //// wdb_update_agent_multi_group success
     //// wdb_delete_agent_belongs success
```
Comment (pereyra-m) [inline src/unit_tests/wazuh_db/test_wdb_agent.c:3391]: These comments have extra `//`, they may need a review

File: src/unit_tests/wazuh_db/test_wdb_agent.c [hunk 3]
```diff
@@ -4217,6 +4342,463 @@ void test_wdb_update_groups_error_json(void **state) {
...
+void test_wdb_update_groups_error_max_path(void **state) {
+    int ret = 0;
+    cJSON *root = NULL;
+    cJSON *row1 = NULL;
+    cJSON *row2 = NULL;
+    cJSON *str1 = NULL;
+    cJSON *str2 = NULL;
+    char *very_long_name = NULL;
+
+    // Generating a very log group name
```
Comment (pereyra-m) [inline src/unit_tests/wazuh_db/test_wdb_agent.c:4354]: Typo: `log` instead of **long**

**Review by pereyra-m — APPROVED** [review]
LGTM!

File: src/unit_tests/wazuh_db/test_wdb_agent.c [hunk 4]
```diff
@@ -755,28 +606,153 @@ void test_wdb_insert_agent_success(void **state)
...
+    expect_string(__wrap_wdbc_query_ex, query, query_str);
+    expect_value(__wrap_wdbc_query_ex, len, WDBOUTPUT_SIZE);
+    will_return(__wrap_wdbc_query_ex, response);
+    will_return(__wrap_wdbc_query_ex, OS_SUCCESS);
+
+    // Parsing Wazuh DB result
+    expect_any(__wrap_wdbc_parse_result, result);
+    will_return(__wrap_wdbc_parse_result, WDBC_OK);
+
+    // Hnadling result and creating agent database
```
[the remark this answers is not in the document]
Comment (DProvinciani) [reply src/unit_tests/wazuh_db/test_wdb_agent.c:619]: Fixed.

File: src/unit_tests/wazuh_db/test_wdb_agent.c [hunk 5]
```diff
@@ -1888,11 +1928,13 @@ void test_wdb_remove_agent_error_socket(void **state)
...
 
     // Calling Wazuh DB
-    expect_value(__wrap_wdbc_query_ex, *sock, -1);
+    expect_any(__wrap_wdbc_query_ex, sock);
     expect_string(__wrap_wdbc_query_ex, query, query_str);
     expect_value(__wrap_wdbc_query_ex, len, WDBOUTPUT_SIZE);
+    will_return(__wrap_wdbc_query_ex, response);
     will_return(__wrap_wdbc_query_ex, OS_INVALID);
 
     // Hnadling result
```
[the remark this answers is not in the document]
Comment (DProvinciani) [reply src/unit_tests/wazuh_db/test_wdb_agent.c:1940]: Fixed

File: src/unit_tests/wazuh_db/test_wdb_agent.c (hunk 1 above)
Comment (DProvinciani) [reply src/unit_tests/wazuh_db/test_wdb_agent.c:3019]: Fixed by updating the branch

File: src/unit_tests/wazuh_db/test_wdb_agent.c (hunk 3 above)
Comment (DProvinciani) [reply src/unit_tests/wazuh_db/test_wdb_agent.c:4354]: Fixed

**Pull request merged**